py/examples/event_recorder/service_config.json

```
@@ -1,9 +1,5 @@
 {
     "name": "recorder",
```

```suggestion
    "name": "amiga",
```

This name no longer matters, but I wonder if we should be consistent and always call it amiga?

---

Does it matter how it's registered in the topic_request_manager tree? I wouldn't mind changing it here, and standardizing it to "amiga service" does make sense. But if we do that, I think we should update all examples that are now hooked up to the amiga service...

---

> Does it matter how it's registered in the topic_request_manager tree?
Not that I am aware of

>  But if we do that, I think we should update all examples that are now hooked up to the amiga service...
Yeah agreed. If you want to do "amiga service" instead of "amiga", you might have to add that as another special case in the `topic_from_uri` method

---

Sorry, I didn't mean to suggest we change the service name in the configs to "amiga service" instead of "amiga".
I just meant to say we should update the "gps" and "oak/*" examples to "amiga", since those are now part of the "amiga" service

---

Agreed

---

I tried changing the service name to `amiga`. Here's what happened:
- Request didn't match (it was registered in the topic tree as `recorder/start`) and it came as `/start` (so, the service name does matter)
- I modified the registration to `/start` --> Couldn't even start the service because cargo was complaining that trailing slashes are forbidden
- I modified the registration to `start` and the python call to `start` --> Request didn't match (expected `/start`)
- I modified the python call to `/start`:
```bash
[ERROR farm_ng_stream::service::event_grpc] Error processing request:
Invalid Key Expr `/start`: empty chunks are forbidden, as well as leading and trailing slashes at 
/cargo/registry/src/index.crates.io-6f17d22bba15001f/zenoh-keyexpr-0.10.1-rc/src/key_expr/borrowed.rs:356.
```

For the time being, I will leave the service name as `recorder`. Happy to look over it again after the 2.3 release.

---

Sorry I guess I should have been more explicit. You would do `recorder/start` in the requestreply method, but not a big deal. The `topic_from_uri` method ignores the name if it is `amiga` (and in a few other cases), so it was wrong to say it doesn't matter. But it does not need to matter, since we can specify with the path

---

The advantage to specifying the service name as `amiga` and putting the device / sub-service in the path is that a single `EventClient` could stream `oak/0/rgb`, `cambus/twist`, call `recorder/start`, etc.

---

This is a pretty nice advantage that I would consider as it might be useful for release 2.3

---

Got it! And, I agree!
Proposed changes applied here: c9a418f78456ccb0ec98b8aafee835fff186b6de